frontend/src/charts/bar.ts

```diff
@@ -56,26 +56,66 @@ const bar_validator = array(
   })
 );
 
+/** Calculate the currencies to use for the chart. */
+function calculateCurrenciesToShow(
+  data: {
+    budgets: Record<string, number>;
+    balance: Record<string, number>;
+  }[],
+  operatingCurrencies: string[]
+): string[] {
+  // Count the usage of each currency in the data.
+  const inData = data.reduce((map, interval) => {
+    Object.keys(interval.budgets).forEach((c) => {
+      map.set(c, (map.get(c) ?? 0) + 1);
+    });
+    Object.keys(interval.balance).forEach((c) => {
+      map.set(c, (map.get(c) ?? 0) + 1);
+    });
+    return map;
+  }, new Map<string, number>());
+
+  const toShow = [];
+  // Always take operating currencies if they are used in the data.
+  for (const currency of operatingCurrencies) {
+    if (inData.has(currency)) {
+      toShow.push(currency);
+      inData.delete(currency);
+    }
```

Comment:
I'm not sure whether the order of currencies is meaningful. If not, we can simply set infinity to all used operating currencies.

Reply:
It's not very important but we should try to have the order be mostly consistent. Sticking to the order of the operating currencies makes sense IMHO